presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -5054,11 +5055,24 @@ public void testLargeIn()
     }
 
     @Test
-    public void testNullOnLhsOfInPredicateDisallowed()
+    public void testNullOnLhsOfInPredicateDisallowedInLegacyMode()
     {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(LEGACY_SEMI_JOIN, "true")
+                .build();
+
         String errorMessage = "\\QNULL values are not allowed on the probe side of SemiJoin operator\\E.*";
-        assertQueryFails("SELECT NULL IN (SELECT 1)", errorMessage);
-        assertQueryFails("SELECT x FROM (VALUES NULL) t(x) WHERE x IN (SELECT 1)", errorMessage);
+        assertQueryFails(session, "SELECT NULL IN (SELECT 1)", errorMessage);
+        assertQueryFails(session, "SELECT x FROM (VALUES NULL) t(x) WHERE x IN (SELECT 1)", errorMessage);
+    }
+
+    @Test
+    public void testNullOnLhsOfInPredicateAllowed()
+    {
+        assertQuery("SELECT NULL IN (1, 2, 3)", "SELECT NULL");
+        assertQuery("SELECT NULL IN (SELECT 1)", "SELECT NULL");
+        assertQuery("SELECT NULL IN (SELECT 1 WHERE FALSE)", "SELECT FALSE");
+        assertQuery("SELECT x FROM (VALUES NULL) t(x) WHERE x IN (SELECT 1)", "SELECT 33 WHERE FALSE");
```

Comment:
`33` was intentionally different from `1` -- there is `1` elsewhere in the query